Registration/RegisterWithAzure.psm1

```diff
@@ -1402,98 +1403,189 @@ function DeActivate-AzureStack{
     } while ($currentAttempt -lt $maxAttempt)
 }
 
+function Get-AzToken
+{
+    [CmdletBinding(DefaultParameterSetName='default')]
+    param
+    (
+        # The Azure PowerShell context representing the context of a token to be resolved.
+        [Parameter()]
+        [ValidateNotNull()]
+        [Microsoft.Azure.Commands.Common.Authentication.Abstractions.IAzureContext] $Context = (Get-AzContext -ErrorAction Stop),
+
+        # The target resource for which a token should be resolved.
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string] $Resource = ($Context.Environment.ActiveDirectoryServiceEndpointResourceId),
+
+        # The target tenantId in which a token should be resolved.
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string] $TenantId = ($t = if ($Context.Tenant) {$Context.Tenant} else {$Context.Subscription.TenantId}),
+
+        # The account for which a token should be resolved.
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string] $AccountId = ($Context.Account.Id),
+
+        # Indicates that target token should be resolved from existing cache data (including a refresh token, if one is available).
+        [Parameter(Mandatory=$true, ParameterSetName='FromCache')]
+        [switch] $FromCache,
+
+        # Indicates that all token cache data should be returned.
+        [Parameter(ParameterSetName='FromCache')]
+        [switch] $Raw
+    )
+
+    $originalErrorActionPreference = $ErrorActionPreference
+    try
+    {
+        $ErrorActionPreference = 'Stop'
+
+        Write-Verbose "Attempting to retrieve a token for account '$AccountId' in tenant '$TenantId' for resource '$Resource'..."
```

Comment:
Rest of the file uses Log-* instead of Write-*

Reply:
This is part of Get-AzToken function which will be moved to common module later.